spec/lib/sanitize_config_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require Rails.root.join('app', 'lib', 'sanitize_config.rb')
+
+describe Sanitize::Config do
+  describe '::MASTODON_STRICT' do
+    subject { Sanitize::Config::MASTODON_STRICT }
+
+    it 'converts h1 to p' do
+      expect(Sanitize.fragment('<h1>Foo</h1>', subject)).to eq '<p>Foo</p>'
+    end
+
+    it 'converts li to p' do
```

Comment:
did i misunderstand the job of UNSUPPORTED_ELEMENTS_TRANSFORMER? to me it read like (h1 h2 h3 h4 h5 h6 blockquote pre li) would be preserved, and anything else would be converted to p

the reason i ask is because converting li to p would insert margins below it, and that causes excessive spacing. essentially, an li element has only 1 visual line break, whereas a p element has 2 visual line breaks.

Reply:
Yes, you probably misunderstood. They all get converted to `<p>`.

I feel like single line breaks can be very hard to notice within longer text, so while for single-line list items it would be best to use `<br />`, `<p>` is more fit for anything that has its own wrapping.

Reply:
ah, so it's like this then?
1. keep (p br span a)
2. convert (h1 h2 h3 h4 h5 h6 blockquote pre li) to p
3. strip all other elements

the "unless" was throwing me off.

i still think semantically if you were trying to map ol/ul and li, then li would be most like a br element. most formatting on the rest of the internet (and indeed most browsers' default css) treats lists as single-break, with double-break around the list due to margin css.

i'm still not sure what the point is in explicitly stripping out the list elements, since they render unobtrusively anyway. my idea of converting to p was more for unsupported elements like marquee. as far as i can tell this current code just removes marquee entirely, right? that would cause multiple marquees to become a run-on sentence.

Reply:
Actually, I think converting `li` to `p` elements might indeed be a problem. For a start, `li` elements can contain a `p` already. Also, regarding how it would be displayed:

## With my patch

![image](https://user-images.githubusercontent.com/384364/58411528-c6b7b280-8074-11e9-9768-7a4529c7cc55.png)

## Converted as `p` tags

![image](https://user-images.githubusercontent.com/384364/58411584-e3ec8100-8074-11e9-863e-eb01cf380d7a.png)

Reply:
likewise, this is what the following markup looks like without any patches:
```
<ul><li>item</li><li>item</li><li>item</li><li>item</li><li>item</li></ul>
```
![image](https://user-images.githubusercontent.com/10606431/58441871-59734400-80ab-11e9-88ee-411241749d69.png)